sample/lib/service/post.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:GTUBT/models/api_response.dart';
+import 'package:GTUBT/models/post.dart';
+import 'package:GTUBT/service/base.dart';
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+
+class PostService extends BaseService {
+  static final PostService _postService = PostService._internal();
+  final servicePath = 'posts';
+  PostService._internal();
+
+  factory PostService() {
+    return _postService;
+  }
+
+  Future<List<Post>> getAll() async {
+    String url = '$baseUrl/$endpointPrefix/$servicePath';
+
+    final response = await http.get('$url');
+
+    if (response.statusCode == 200) {
+      final apiResponse = ApiResponseList<Post>.fromJson(json.decode(response.body));
+      if (apiResponse.status == 200) {
+        return apiResponse.body;
+      }
+    }
+
+    return List<Post>();
+  }
+
+  Future<Post> get(String id) async {
+    String url = '$baseUrl/$endpointPrefix/$servicePath/$id';
+
+    final response = await http.get('$url');
+
+    if (response.statusCode == 200) {
+      final apiResponse = ApiResponseSingle<Post>.fromJson(json.decode(response.body));
+      if (apiResponse.status == 200) {
+        return apiResponse.body;
+      }
+    }
+
+    return null;
+  }
+
+  Future<http.Response> post(Post post) async {
+    String url = '$baseUrl/$endpointPrefix/$servicePath';
+    var postInJson = post.toJson();
+    var bodyData = json.encode(postInJson);
+
+    final response = await http.post(
+      '$url',
+      headers: <String, String> {
+        'Content-Type': 'application/json; charset=UTF-8',
+      },
+      body: bodyData
+    );
+
+    return response;

```

Comment:
return etmeden once request basarili mi diye kontrol etmek gerekli sanirim?

Reply:
Bu metod'un gorevi POST requesti atmak oldugu icin return degeri olarak HTTP response veriyoruz. Obur turlu eger request basariliysa true degilse false donmek gibi bir sey yaparsak metodun semantigi ile ortusmeyecegini dusunuyorum. Metodu cagiran taraf yaptigi cagridan sonra islemin basarili olup olmadigini status code ile kontrol ederek arayuzu guncelleyebilir. Boylece service methodlarimiz arasindaki butunluk de bozulmamis olur.

Reply:
Bunu niye resolved olarak isaretledin anlamadim. 

Buna ben approved dedim ama o kadar kasmaya gerek gormemistim su noktada servis cevaplarina. Ama servisin mantigina uygun hareket etmek istiyorsak aksine servisin verdigi cevap raw http response olmamali. Servis o katmani abstractlastirmali. true false daha iyi olabilir. Hatta en iyi yontem bizim response type imizi donup icerisinde false ise neden false olduguna dair mesaj da barindirmasi olur. Boylelikle bi alan yanlissa o alani invalid olarak marklayabiliriz ui da. 

Ama yine de bu kisimlarin ucu cok acik. Gurolla sana birakiyorum karari.

Reply:
O zaman şöyle yapalım. ServiceResponse diye bir class tanımlayalım template parametreli. Bunun içine status, message ve template birer obje koyalım. HTTP metodu ne olursa olsun tüm servis metodları bu response tipini return etsin. 

Resolve ettim çünkü response yapısını oluşturup elimizdeki servisleri buna uydurmamız başka bir issue'da ele alınırsa daha uygun olur. slack'te konuşup kart açabiliriz. Şimdilik unresolve ettim yine de.

Reply:
burada dogrudan response donuyor olmak benim hosume gitmedi. bunu onurun dedigi sekilde yapmak daha mantikli. bu is icinde yapmayalim madem. @onurccn baska bir is icinde bunu yapar misin?

Reply:
Tamamdir. Yasirin bahsettigi serviceResponse olayini zaten benim event service te ekledigim ApiResponse la cozuyoruz diye dusunuyorum. Onu kullanacak sekilde ayarlariz